purchase_work_acceptance/models/purchase.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2019 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    wa_count = fields.Integer(
+        compute='_compute_wa_ids',
+        string='WA count',
+        default=0,
+    )
+    wa_ids = fields.Many2many(
+        comodel_name='work.acceptance',
+        inverse_name='purchase_id',
+        compute='_compute_wa_ids',
+        string='Work Acceptances',
+    )
+    wa_line_ids = fields.One2many(
+        comodel_name='work.acceptance.line',
+        inverse_name='purchase_line_id',
+        string="WA Lines",
+        readonly=True,
+    )
+
+    def _compute_wa_ids(self):
+        for order in self:
+            for line in order.order_line:
+                order.wa_ids |= line.wa_line_ids.mapped('wa_id')
+            order.wa_count = len(order.wa_ids)
+
+    @api.multi
+    def action_view_wa(self):
+        act = self.env.ref('purchase_work_acceptance.action_work_acceptance')
+        result = act.read()[0]
+        create_wa = self.env.context.get('create_wa', False)
+        result['context'] = {
+            'default_purchase_id': self.id,
+            'default_partner_id': self.partner_id.id,
+            'default_company_id': self.company_id.id,
+            'default_date_due': self.date_planned,
+            'default_wa_line_ids': [(0, 0, {
+                'purchase_line_id': line.id,
+                'name': line.name,
+                'product_uom': line.product_uom.id,
+                'product_id': line.product_id.id,
+                'price_unit': line.price_unit,
+                'product_qty': max(line.product_qty - line.qty_invoiced,
+                                   line.product_qty - line.qty_received,),
+                }) for line in self.order_line],
+        }
+        if len(self.wa_ids) > 1 and not create_wa:
+            result['domain'] = "[('id', 'in', " + str(self.wa_ids.ids) + ")]"
+        else:
+            res = self.env.ref(
+                'purchase_work_acceptance.view_work_acceptance_form', False)
+            result['views'] = [(res and res.id or False, 'form')]
+            if not create_wa:
+                result['res_id'] = self.wa_ids.id or False
+        return result
+
+    @api.multi
+    def action_view_invoice(self):
+        create_bill = self.env.context.get('create_bill', False)
+        if create_bill:
```

Comment:
Should we make it shorter like this?
```suggestion
        if self._context.get('create_bill', False):
```

Reply:
I will changed.